### ISSUE TITLE
Weekly Log for Shahar Yair

### ISSUE THREAD

**Issue by TomPalny** [TechnionYP5777/SmartParking#4]
# The Weekly Log
New for semester 2, a fresh new page with a different name.
This is meant to help in monitoring everyone's assignments in the projects and what they are currently involved with. 

Please use this as a weekly log from week 2 till the end of semester, both for you and me to see.
Log here *_past_* work that is finished, as well as *_current_* tasks you are doing.

Whenever possible, please include an issue number that you are working on, or that you have finished working on.
Best to use the bulleted list as a way to put things organized. 

**Have a great semester!**

**Comment by Shahar-Y** [comment]
Week 1:
This week I started working on creating the log files for when the code throws an exception: issue #11.
For now, a log file is created under "logs" folder in the project, containing the stack trace. 
Apparently, a more detailed log is required so I'll keep working on it in week 2, hopefully finishing it.
Next week plans:
-  Continue working on the log files creation as requested.
-  Working with David on transferring the code to 'ionic' development tool: issue #21.

**Comment by TomPalny** [comment]
Hi Shahar, please reference issue numbers to all the things you're working on that relate to them. If no issue is relevant, for example, if you're learning a new technology such as ionic, that's fine. Anything that does relate to an issue should have the number mentioned. Thanks

**Comment by Shahar-Y** [comment]
Week 2: 
During this week, David and I started working on installing ionic (#21) into my laptop in order to create a basic project and start integrating ionic to the project's gui. Much of the credit should go to Yael who helped us a great deal with her instructions.
Next week plans:
-  I intend to continue making progress with the ionic integration. (#21)
- Working on the log files, which I slightly neglected because of ionic's time consumption. (#11)

**Comment by Shahar-Y** [comment]
Week 3:
Haven't got the chance to write the weekly log yet because of the holiday, sorry.
On the third week of the semester, I continued working with David on the Ionic integration (#21). With the help of Zahi we figured out the problem Travis was having with adding REST to the project and fixed it.
I also created a wiki page for Ionic (#26): https://github.com/TechnionYP5777/SmartParking/wiki/Ionic .
At the last group meeting we held, it was decided that issue #11 about the log files will be postponed until Ionic installation is completed because it is more urgent, and because it relies on Ionic.
Next week plan is to keep pushing with David on Ionic.

**Comment by TomPalny** [comment]
Hi Shahar,
Please write the current week's work when you have a chance and reference the issue number you're working on. We are now in a the stage where the integration with ionic should be complete, or nearly complete and we need to cover more ground.

**Comment by Shahar-Y** [comment]
## Weeks 4+5: 
### My Progress:
 - I kept working with David on Ionic (closed  issue #21 and started with #34). After some serious struggle along with Yael, we managed to establish a connection to the server and present data from it to the ionic app. We can pull data from websites in Json format, including our localhosts. So now we have both  client side and server side with a very basic functionality.
 - Closed #26 about the wiki page.
 - Making the presentation for the meeting in 24/4 (#28).

### Next Week:
- [x] Create authentication of the user, probably with Zahi's GUI.
- [x] Checking the functionality available for us with the users.
- [x] Dealing with connection errors that could be on the way.

**Comment by Shahar-Y** [comment]
# Week 6:
## My Progress:
 - Fixed the bug presented in issue #38 with David. Ionic now works on all computers.
 - Connected the java code to ionic services (#34) in order to get data from the server, and upload it to the localhost.
 - Created authentication with David, (#37). It works, but we still have a problem with presenting the data.
 - Opened issue #39 for learning the SpringBoot web services.

## Next Week:
 - [x] Finish with the Login & Register page (#37).
 - [ ] Work on "My Details" page (an issue will be opened).
 - [x] Work on whichever ionic-related problem that would present itself in the near future.

**Comment by Shahar-Y** [comment]
# week 7
## My Progress 

- Worked on the location services with David and Shay. (#42)
- Worked on and finished with the Register service. (#43)
- Continue learning Spring Boot services. (#39)
- Worked on the Login page with David. (#37)

## Next week:

- Keep working on the User services.
- Work on the log files.

**Comment by TomPalny** [comment]
You're neglecting the weekly log page. This is one of the biggest tools we are going to use to gauge your contribution to the project at the end of semester. Don't forget to link to issues/commits that are relevant on each task.

**Comment by TomPalny** [comment]
Also, write here which weeks you are not working (in total we have 13 weeks in the semester, and the log is only showing till week 7, while we are already on week 10)

**Comment by Shahar-Y** [comment]
You're right. I forgot to write the weekly log before I started the pagra. 

In **week 8** I kept working with David on the ionic services, which includes #37, #39 and #43. 
The things I currently need to finish mainly include finishing up with the login service (which works, but not fully since it doesn't show the details of the logging in, though they are in the system, leading the user to the relevant page or show errror in case of an error). Also, the same thing goes with the register service- it creates a new user in parse but doesn't give any feedback. 
In order to try and finish both of these things before the demo, I formally request to finish my pagra early and start working as soon as tomorrow. Hopefully, it won't be a problem since I haven't made any commits in the last three days before starting the pagra. 
## Weeks 9-10: PAGRA

**Comment by TomPalny** [comment]
sure, no problem at all! thank you for the detailed info

בתודה,
תום

**Comment by Shahar-Y** [comment]
# Week 11
On week 11 I worked on fixing the login and register functionality for the demo. 
Also, I started working on fixing the bugs as we agreed upon after the demo #59.
Currently I am working on the logout functionality, and after that I'll work on changing the login functionality to work with username and password instead of with car number, get feedback from the register, and thinning out the user details as agreed with Tanya.

**Comment by Shahar-Y** [comment]
# Week 12
This week I closed all of the sub- issues of issue #59. In hindsight, perhaps I should have spread it across several issues so that it would be more organized.
Issue #59 included:

- Transferring the user to the right page after logging in and registering (easy)
- Automatically logging the user in after register (easy).
-  Adding logout functionality.
- Returning feedback after registerring.

Both of the latter took most of the time and commits for this issue.
Also, I forgot that the logout was shared with issue #43 so YAY two birds...
 # Next week 
I'm going to work on the logger and doing some patching-up for my entire part of the work so it will be ready for submittion.
Also, creating "My Details" page which should be quite easy.

**Comment by Shahar-Y** [comment]
# Week 13
this week I worked on these issues: 
- #59 : Fixing last sprint's bugs, as Tanya instructed and as agreed with the group. **- closed.**
- #64 : Added MyDetails page and got it to show the user details. **- closed.**
- #43 : Working on the user services: finished all of the tasks in there but: **- closed**
- #76 : Get last parking slot: still need to get the last parking slot of the user.
- #77 : Created MyDetailsEdit page, but the functionality is still incomplete (though most of it is ready).
- #81 : Bug in register: Apperenately, the register stopped working for a while for some reason. I improved its feedback so now the user will know what exactly went wrong. I believe it is fixed, but still waiting to Shay -who found the bug- to approve it.
Also, worked on the poster #69 

# Next Week
On the last week of the semester, everything must be competed. that includes:
- [x] #76.
- [x] #77.
- [x] #81 - As I said, I believe it is done, but waiting for response from Shay.
- [x] Other things I can't remember right now because I'm tired (To be continued).

**Comment by Shahar-Y** [comment]
# Last Week of the Semester
The last week was very productive, and included two consecutive white nights with David, and the second also with Shay 😴 .
I've added new functionality and services to the application, 
and fixed many of the bugs that came up as the project came to its ending.
On the last week I worked on the following issues: (closed by this order)

- Bug in register: #81.
- Chrome problem: #91. - Fixed by Shay.
- Added identification for each user, so that more than one user would be able to log in. #92.
- Login bug #103. 
- Fix menu options: #104. 
- Creating MyDetailsEdit page. #77.
- No internet connection feedback bug: #93.
- Hiding login/logout from menu: #97. Similar to #104.
- Bug in Edit Details sticker. #113.
- Get last parking slot for the user details: #76.
- Another problem with the menu: #116.
- Logout stopped working due to some changes made. It had to be fixed. #120.
- Spinner added to the waiting in the services #112.
- Refactoring & Coding standrts. #74.
- Other errors explained in #105.

**Comment by Shahar-Y** [comment]
# Reality Testing

On the 2-3.7.17 I worked on fixing the following issues that were opened: 
#137 , #138, #147 #149 . All were about the login/register/editDetails functionalities, and all were fixed within the timeframe.
Also, opened #143 Of the announcements.

I had a great time working on this project, and I have to say im a little sad to see it come to an end 😢 .
Love y'all! @DavidCohen55 @shaysegal @zahimizrahi @sefialbo @TomPalny  ❤️ 
![image](https://media.giphy.com/media/8UF0EXzsc0Ckg/giphy.gif)
